### ISSUE TITLE
False positive CA2252 despite `EnablePreviewFeatures`

### ISSUE THREAD

**Issue by lostmsu** [dotnet/roslyn-analyzers#6484]
### Analyzer

**Diagnostic ID**: [CA2252](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2252): `Using both 'static' and 'abstract' modifiers requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0 (Latest)

Also tried without the package

### Describe the bug

Building the attached project results in `CA2252` error in `TypeNum.INumeral` interface definition.

### Steps To Reproduce

1. Download the attached project
2. Run `dotnet build` (Visual Studio 17.4.4 also fails)

### Expected behavior

No errors

### Actual behavior

```
INumeral.cs(6,29,6,32): error CA2252: Using both 'static' and 'abstract' modifiers requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.

INumeral.cs(6,35,6,38): error CA2252: Using both 'static' and 'abstract' modifiers requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.
```

## Additional context
```
➜ dotnet --info
.NET SDK:
 Version:   7.0.102
 Commit:    4bbdd14480

Host:
  Version:      7.0.2
  Architecture: x64
  Commit:       d037e070eb

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.203 [C:\Program Files\dotnet\sdk]
  5.0.214 [C:\Program Files\dotnet\sdk]
  5.0.303 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.100-rc.2.21505.57 [C:\Program Files\dotnet\sdk]
  6.0.308 [C:\Program Files\dotnet\sdk]
  7.0.102 [C:\Program Files\dotnet\sdk]
```

**Comment by Youssef1313** [comment]
What is the `TargetFramework`(s) you are building against?

**Comment by lostmsu** [comment]
I see, thanks.

Is it possible to build using .NET 6 SDK when .NET 7 SDK is also installed? Would you recommend against it and just drop .NET 6 support if I need this feature?

**Comment by manfred-brands** [comment]
> What is the `TargetFramework`(s) you are building against? If this error happens for net6.0 builds on the .NET 7 SDK, then this is by-design.

Our libraries are multi-targeting, so target both NET6.0 and NET7.0.
Regardless if we specify `<EnablePreviewFeatures>true</EnablePreviewFeatures>` or not, the NET6.0 target doesn't compile.
If we do specify it, the `RequiresPreviewFeatureAttribute` gets added to the NET7.0 target where it shouldn't be.

We tried conditionally setting this based upon TargetFramework and disabled CA2252 as it fires unexpectedly.
I then would expect the attribute to be in the NET6 generated AssemblyInfo.cs but it isn't.

I targeting a framework that doesn't support it, I get _CS8919: Target runtime doesn't support static abstract members in interfaces_. Doesn't that mean that CA2252 is not necessary here?
